### PR TITLE
🐛 Fix infinite scroll for shared with me

### LINKED ISSUE #637: The infinit scroll doesn't work in the "Shared with me" section
It displays the same files all the time and never finish

### PR DESCRIPTION
**Pull request by MontaGhanmy**

### CODE REVIEW

**Review by ericlinagora — APPROVED** [review]

File: tdrive/backend/node/src/services/documents/services/index.ts [hunk 1]
```diff
@@ -112,7 +112,7 @@ export class DocumentsService {
     context: DriveExecutionContext & { public_token?: string },
```
Comment (shepilov) [inline tdrive/backend/node/src/services/documents/services/index.ts:109]: could you add tests please

File: tdrive/backend/node/src/core/platform/services/search/adapters/mongosearch/index.ts [hunk 2]
```diff
@@ -206,11 +206,23 @@ export default class MongoSearch extends SearchAdapter implements SearchAdapterI
...
 
-    const { query, sort, project } = buildSearchQuery<EntityType>(entityType, filters, options);
+    const {
+      query,
+      sort: sortFields,
+      project,
+    } = buildSearchQuery<EntityType>(entityType, filters, options);
 
     logger.info(`Search query: ${JSON.stringify(query)}`);
     console.log(query);
```
Comment (ericlinagora) [inline tdrive/backend/node/src/core/platform/services/search/adapters/mongosearch/index.ts:212]: beurk

File: tdrive/backend/node/test/e2e/documents/documents-pagination-sorting.spec.ts [hunk 3]
```diff
@@ -1,6 +1,7 @@
 import { describe, beforeAll, afterAll, it, expect } from "@jest/globals";
 import { init, TestPlatform } from "../setup";
 import UserApi from "../common/user-api";
+import { async } from "rxjs";
```
Comment (ericlinagora) [inline tdrive/backend/node/test/e2e/documents/documents-pagination-sorting.spec.ts]: aggressive IDE

File: tdrive/backend/node/test/e2e/documents/documents-pagination-sorting.spec.ts [hunk 4]
```diff
@@ -152,5 +153,55 @@ describe("The Documents Browser Window and API", () => {
...
       expect(isSorted).toBe(true);
     });
+
+    it("Should paginate shared with me ", async () => {
+      const sharedWIthMeFolder = "shared_with_me";
+      const oneUser = await UserApi.getInstance(platform, true, { companyRole: "admin" });
+      const anotherUser = await UserApi.getInstance(platform, true, { companyRole: "admin" });
+      let files = await oneUser.uploadAllFilesOneByOne();
+      for (const file of files) {
+        await anotherUser.shareWithPermissions(file, oneUser.user.id, "read");
```
Comment (ericlinagora) [inline tdrive/backend/node/test/e2e/documents/documents-pagination-sorting.spec.ts]: i'm not sure what's going on here, `oneUser` uploads the files, then, `anotherUser` shares them with `oneUser` ?

File: tdrive/backend/node/test/e2e/documents/documents-pagination-sorting.spec.ts [hunk 5]
```diff
@@ -152,5 +153,55 @@ describe("The Documents Browser Window and API", () => {
...
+      expect(docs.children.length).toBeLessThanOrEqual(limit);
+    });
+
+    it("Should sort shared with me by name in ascending order", async () => {
+      const sharedWIthMeFolder = "shared_with_me";
+      const oneUser = await UserApi.getInstance(platform, true, { companyRole: "admin" });
+      const anotherUser = await UserApi.getInstance(platform, true, { companyRole: "admin" });
+      let files = await oneUser.uploadAllFilesOneByOne();
+      for (const file of files) {
+        await anotherUser.shareWithPermissions(file, oneUser.user.id, "read");
```
Comment (ericlinagora) [inline tdrive/backend/node/test/e2e/documents/documents-pagination-sorting.spec.ts]: ditto

File: tdrive/backend/node/test/e2e/documents/documents-pagination-sorting.spec.ts (hunk 3 above)
Comment (MontaGhanmy) [reply tdrive/backend/node/test/e2e/documents/documents-pagination-sorting.spec.ts]: yess @ericlinagora

File: tdrive/backend/node/test/e2e/documents/documents-pagination-sorting.spec.ts (hunk 4 above)
Comment (MontaGhanmy) [reply tdrive/backend/node/test/e2e/documents/documents-pagination-sorting.spec.ts]: It should be `oneUser` shareing with `anotherUser`, fixed it!

File: tdrive/backend/node/src/core/platform/services/search/adapters/mongosearch/index.ts (hunk 2 above)
Comment (MontaGhanmy) [reply tdrive/backend/node/src/core/platform/services/search/adapters/mongosearch/index.ts:212]: Had to be done so it doesn't conflict with the actual `sort` const, but I refactored the code so it's not there anymore.

Comment (ericlinagora) [reply tdrive/backend/node/src/core/platform/services/search/adapters/mongosearch/index.ts:212]: i meant the `console.log`

**Review by shepilov — APPROVED** [review]

**Pull request merged**